cargo-dist/src/backend/ci/github.rs

```diff
@@ -34,19 +37,43 @@ struct CiTask {
 
 /// Generate CI for Github
 ///
-/// This actually creates a file and writes to disk!
-pub fn generate_github_ci(dist: &DistGraph) -> Result<(), miette::Report> {
-    const GITHUB_CI_DIR: &str = ".github/workflows/";
-    const GITHUB_CI_FILE: &str = "release.yml";
-
-    // FIXME: should we try to avoid clobbering old files..?
-    let ci_dir = dist.workspace_dir.join(GITHUB_CI_DIR);
-    let ci_file = ci_dir.join(GITHUB_CI_FILE);
+/// This generates the requested configuration and returns it as a string.
+pub fn generate_github_ci(dist: &DistGraph) -> Result<String, miette::Report> {
     let ci_info = compute_ci_info(dist);
 
     let rendered = dist
         .templates
         .render_file_to_clean_string(TEMPLATE_CI_GITHUB, &ci_info)?;
+
+    Ok(rendered)
+}
+
+fn github_ci_path(dist: &DistGraph) -> camino::Utf8PathBuf {
+    let ci_dir = dist.workspace_dir.join(GITHUB_CI_DIR);
+    ci_dir.join(GITHUB_CI_FILE)
+}
+
+/// Generate CI for Github
+///
+/// This checks whether the new configuration differs from the config on disk
+/// writhout actually writing the result.
+pub fn check_github_ci(dist: &DistGraph) -> Result<bool, miette::Report> {
+    let ci_file = github_ci_path(dist);
+
+    let rendered = generate_github_ci(dist)?;
+    // FIXME: should we catch all errors, or only LocalAssetNotFound?
+    let existing = LocalAsset::load_string(ci_file).unwrap_or("".to_owned());
+    Ok(rendered == existing)
```

Comment:
Hmm, feels like this probably wants to be just returning `Result<(), ...>` and producing the DistError::CheckFileMismatch itself, so that paths/details can be given?